File: Core/OBJECT/Sensore.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.OBJECT
{
    public class Sensore
    {
        public Sensore()
        {
            STATO = StatoSensore.DISATTIVO;
        }
        public int ID { get; set; }
        public string NAME { get; set; }
        public int STATO { get; set; }

        public void CaricaComandi(List<string> aComandi)
        {
            CmdAccendi = aComandi.ElementAt(0);
            RitornoAccendi = aComandi.ElementAt(1);
            CmdSpegni = aComandi.ElementAt(2);
            RitornoSpegni = aComandi.ElementAt(3);
            CmdIngresso = aComandi.ElementAt(4);
            IngressoAlto = aComandi.ElementAt(5);
            IngressoBasso = aComandi.ElementAt(6);
        }

        private string CmdAccendi { get; set; }
        private string RitornoAccendi { get; set; }
        private string CmdSpegni { get; set; }
        private string RitornoSpegni { get; set; }
        private string CmdIngresso { get; set; }
        public string IngressoAlto { get; private set; }
        public string IngressoBasso { get; private set; }

        public string Accendi()
        {
            return CmdAccendi;
        }
        public string Spegni()
        {
            return CmdSpegni;
        }
        public string Ingresso()
        {
            return CmdIngresso;
        }

    }
}

File: ReactionTrainingPY/it/napalm/core/object/Sensore.py
'''
Created on 18/mar/2015

@author: Luigi
'''
from it.napalm.core.object.StatoSensore import StatoSensore
class Sensore(object):
    ID = 0
    NAME = ''
    STATO = 0

    def __init__(self):
        self.STATO = StatoSensore.DISATTIVO
        
    def CaricaComandi(self, aComandi):
        self.CmdAccendi = aComandi[0]
        self.RitornoAccendi = aComandi[1]
        self.CmdSpegni = aComandi[2]
        self.RitornoSpegni = aComandi[3]
        self.CmdIngresso = aComandi[4]
        self.IngressoAlto = aComandi[5]
        self.IngressoBasso = aComandi[6]
        
    CmdAccendi = ''
    RitornoAccendi = ''
    CmdSpegni = ''
    RitornoSpegni = ''
    CmdIngresso = ''
    IngressoAlto = ''
    IngressoBasso = ''
    
    def Accendi(self):
        return self.CmdAccendi
    
    def Spegni(self):
        return self.CmdSpegni
    
    def Ingresso(self):
        return self.CmdIngresso
        